typed-function.js

```diff
@@ -1428,6 +1464,14 @@
       typed.conversions.push(conversion);
     };
 
+    /**
+     * Check if an entity is a typed function created by this instance
+     * @param {any} entity
+     * @returns {boolean}
+     */
+    typed.isTypedFunction = entity => entity && typeof entity === 'function' &&
+      '_isTypedFunction' in entity
```

Comment:
Two (small) thoughts: 

1. I think it would be fine to to check whether `.signatures` is an object instead of introducing a new property `_isTypedFunction` (as in: less code is better in general). If you really find it too risky to rely on `.signatures` I'm ok with it though, no big deal. 
2. There are currently two occurrences in the code which do check whether something is a typed-function: In `find` and in `typed` (the new "// Case 2: Existing typed function"). Currently these validations check against `.signatures`. It would be neat to replace these with using `isTypedFunction` so the behavior is consistent throughout the library.

Reply:
Fair point. Ok let's go for `._isTypedFunction` then 👍

Reply:
Alright, rebased and isTypedFunction uniformly used internally. As far as I know, everything is set with this PR.